### PR TITLE
release process guide

### LINKED ISSUE #911: Release process guide / manual
part of #864 

### What problem are you facing?

There is no release guide that educates and informs an engineer of Crossplane on how to release a new version of the software.  @jbw976 has run the releases so far but no one else knows how to perform one.

### How could Crossplane help solve your problem?

We should create a new release process guide that provides all the instructions and tips for running a release of the Crossplane software.  This guide should cover how to tag/branch, build and publish, and promote.  It should also cover patch releases as well.

### PR DESCRIPTION
**Pull request by jbw976**
### Description of your changes

This PR contains a complete guide for how to run a Crossplane release.  The intended audience is Crossplane contributors that are already familiar with how the project is run.  Anyone should be able to follow this guide start to finish and release a new version of Crossplane as well as all the stacks that the Crossplane community maintains.

Fixes #911 

### Checklist

I have:
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Ensured this PR contains a neat, self documenting set of commits.
- [x] Updated any relevant [documentation], [examples], or [release notes].
- [x] Updated the RBAC permissions in [`clusterrole.yaml`] to include any new types.

[skip ci]

[documentation]: https://github.com/crossplaneio/crossplane/tree/master/docs
[examples]: https://github.com/crossplaneio/crossplane/tree/master/cluster/examples
[release notes]: https://github.com/crossplaneio/crossplane/tree/master/PendingReleaseNotes.md
[`clusterrole.yaml`]: https://github.com/crossplaneio/crossplane/blob/master/cluster/charts/crossplane/templates/clusterrole.yaml

### CODE REVIEW

File: docs/release/release-process.md [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+All the details are available in the sections below, but we'll start this guide with a very high
+level sequential overview for how to run the release process.
+
+1. **feature freeze**: Merge all completed features into master branches of all repos to begin
+   "feature freeze" period
+1. **RC tag**: Run tag pipeline to tag the release candidate (RC) build in master for each repo
+1. **branch**: Create a new release branch using the GitHub UI for each repo
+1. **release stacks**: Run the release process for each **stack** that we maintain
+    1. **test** Test builds from the release branch, fix any critical bugs that are found
+    1. **version**: Update all version information in the stack metadata and docs in the release
```
Comment (hasheddan) [inline docs/release/release-process.md]: Do we want to mention the integration test change that has to happen here as well? I can look into fixing that for future releases.

File: docs/release/release-process.md [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+       branch
+    1. **tag**: Run the tag pipeline to tag the release branch with an official semver
+    1. **build/publish**: Run build pipeline to publish build with official semver
+1. **docs/examples**: In Crossplane's release branch, update all examples and docs to reference the
```
Comment (hasheddan) [inline docs/release/release-process.md]: Same with integration test here :/

File: docs/release/release-process.md [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+
+At this point, we should have a release branch created in all repos with all issues resolved that
+were triaged to be release blockers.  This section will walk through how to release the Stacks
+themselves and does not directly apply to core Crossplane.
+
+In the **release branch** for each Stack, you should update the version tag or metadata in:
+
+* `app.yaml` - `version`
+* `install.yaml` - `image`
+* `integration_tests.sh` - `STACK_IMAGE`
```
Comment (hasheddan) [inline docs/release/release-process.md]: Ah see that it is addressed here. I am fine to leave it out above as anyone running the release will hopefully have read this part as well.

File: docs/release/release-process.md [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+published.  These patch releases are always run from the last release branch, we do **not** create a
+new release branch for a patch release.
+
+The basic flow is **very** similar to a normal release, but with a few less steps.  Please refer to
+details for each step in the sections above.
+
+* Fix any bugs in `master` first and then `cherry-pick -x` to the release branch
+  * If `master` has already removed the relevant code then make your fix directly in the release
+    branch
+* After all testing on the release branch look good, run the `tag` pipeline on the release branch
```
Comment (hasheddan) [inline docs/release/release-process.md]: Versioning updates are also needed here in most cases.

**Comment by jbw976** [comment]
Incorporated your feedback to add "integration tests" version updates more explicitly in the summary and add a version update reminder in the patch release steps. Thanks!

**Pull request merged**